piff/optical_model.py

```diff
@@ -175,52 +256,215 @@ def fit(self, star):
+
+    def params_tokwargs(self,params):
+        """Fill kwarg from params ndarray or list
+
+        :param params               An ndarray or list with parameters in order
+
+        :returns: a dictionary of named parameters
+        """
+        kwargs = {}
+        kwargs['zernike_coeff'] = params[self.idx_z0:self.idx_z0+self.nZ+1]
+        kwargs['r0'] = params[self.idx_r0]
+        kwargs['L0'] = params[self.idx_L0]
+        kwargs['g1'] = params[self.idx_g1]
+        kwargs['g2'] = params[self.idx_g2]
+        return kwargs
+
+    def kwargs_toparams(self,zernike_coeff=[0,0,0,0,0],r0=0.15,L0=0.,g1=0.,g2=0.):
+        """Fill params ndarray from kwargs
+
+        :param zernike_coeff:      A ndarray or list with [0, 0, z2, z3, z4, z5, z6...z11..z37] [default=[0,0,0,0,0]]
+        :param r0:                 Value of r0 for Atmospheric Kernel. [default=0.15]
+        :param L0:                 Value of L0 for Atmospheric Kernel. [default=0.0]
+        :param g1:                 Value of g1 to Shear PSF [default=0.]
+        :param g2:                 Value of g2 to Shear PSF [default=0.]
+
+        :returns: an ndarray of parameters
+        """
+        params = np.zeros(self.param_len)
+
+        # adapt to a zernike_coeff array shorter or longer than 0:37+1
+        zlen = len(zernike_coeff)
+        if zlen==self.nZ+1:
+            params[self.idx_z0:self.idx_z0+self.nZ+1] = zernike_coeff
+        elif zlen<self.nZ+1:
+            params[self.idx_z0:self.idx_z0+zlen] = zernike_coeff
+        elif zlen>self.nZ+1:
+            params[self.idx_z0:self.idx_z0+self.nZ+1] = zernike_coeff[0:selfnZ+1]
+
+        params[self.idx_r0] = r0
+        params[self.idx_L0] = L0
+        params[self.idx_g1] = g1
+        params[self.idx_g2] = g2
+        return params
+
+
+    def getProfile(self,*params,zernike_coeff=None,r0=None,L0=None,g1=None,g2=None):
+        """Get a version of the model as a GalSim GSObject
 
-        if self.g1 is not None or self.g2 is not None:
-            prof = prof.shear(g1=self.g1, g2=self.g2)
+        :param params          An ndarray with the parameters ordered via idx_PARAM data members
+        :param zernike_coeff   A list of the Zernike coefficients in units of [waves]
+        :param r0              Atmospheric Fried parameter [meters]
+        :param L0              Atmospheric Outer scale [meters]
+        :param g1              Atmospheric g1 Shear
+        :param g2              Atmospheric g2 Shear
+
+        :returns: a galsim.GSObject instance of the combined optics, atmosphere and diffusion PSF
+        """
+
+        # decode input, if params array or list is present use it alone,
+        # otherwise parameters are in getProfile argument list
+        if len(params)==1 :
+            zernike_coeff,r0,L0,g1,g2 = self.unpack_params(params[0])
+        elif len(params)!=0 :
+            logger.error("getProfile cannot decode arguments:",params)
```

Comment:
I think this should raise an exception (ValueError probably), not just write something to the log.